lib/src/connection_fsm.dart

```diff
@@ -12,6 +12,17 @@ abstract class _PostgreSQLConnectionState {
   }
 
   _PostgreSQLConnectionState onMessage(ServerMessage message) {
```

Comment:
Since notifications don't have quite the same lifecycle or need for the finite state machine, it'd be better to make this `if (message is NotificationResponseMessage)` check in `connection._readData` and avoid the FSM altogether. There is already logic in `connection._readData` to separate out `ErrorResponseMessage`s, so it'll fit right in.